src/menu.rs

```diff
@@ -6,14 +6,57 @@ use std::{
   hash::{Hash, Hasher},
 };
 
+use crate::platform_impl::{MenuBuilder as PlatformMenuBuilder, MenuItem};
+
+#[derive(Debug, Clone)]
+pub struct MenuBuilder(pub(crate) PlatformMenuBuilder);
+
+impl MenuBuilder {
+  pub fn init() -> Self {
+    Self(PlatformMenuBuilder::init())
+  }
+
+  pub fn init_with_title(title: &str) -> Self {
+    Self(PlatformMenuBuilder::init_with_title(title))
+  }
+
+  pub fn add_children(&mut self, children: MenuBuilder, title: &str, enabled: bool) {
+    self.0.add_children(children.0, title, enabled);
+  }
+
+  pub fn add_item(
+    &mut self,
+    menu_type: MenuType,
+    text: &str,
+    keyboard_accelerator: Option<&str>,
+    enabled: bool,
+    selected: bool,
+  ) -> (MenuId, MenuItem) {
+    let menu_id = MenuId::new(&text);
+    let item = self.0.add_custom_item(
+      menu_id,
+      menu_type,
+      text,
+      keyboard_accelerator,
+      enabled,
+      selected,
+    );
+    (menu_id, item)
+  }
+
+  pub fn add_system_item(&mut self, item: SystemMenu, menu_type: MenuType) -> MenuItem {
```

Comment:
I prefer to keep the name of `MenuItem`. Could related types and methods fall back to this accordingly?